### DISCUSSION TITLE
Is discord.ui.Modal supports button?

### DISCUSSION THREAD

**Discussion by beatbox4108** [Rapptz/discord.py#8312] [Q&A] [answered]
I've tried, but it raises an error.
```
discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In data.components.0.components.0: Value of field "type" must be one of (3, 4).
```

This is the minimum code.
```python3
import discord
import discord.app_commands
import discord.ui

token = #TOKEN here
client = discord.Client(intents=discord.Intents.default())
tree = discord.app_commands.CommandTree(client)

@tree.command(
    name="modal",
    description="Send modal."
)
async def modal(ctx:discord.Interaction):
    modal=discord.ui.Modal(title="Hello!")
    modal.add_item(discord.ui.Button(label="Button"))
    await ctx.response.send_modal(modal)

@client.event
async def on_ready():
    print("ready")
    await tree.sync()
client.run(token)
```

How can I fix this? Or Is It not able?

**Comment by bijij** [comment, marked as answer]
Modals only support Select and TextInput components.

**Reply by beatbox4108** [reply]
Okay, thanks!

**Comment by beatbox4108** [comment]
I forgot to close. sorry 🙏

**Comment by lhy-hoyin** [comment]
Update for future developers:

> Support for components in modals is currently limited to type 4 (Text Input).

As mentioned by Discord [here](https://discord.com/developers/docs/interactions/overview#modals) and [here](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-modal), Modal only supports Text Input, so Select and Button will not work.
